website/source/docs/internals/providers/new-resource.html.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: "docs"
+page_title: "Creating Resources"
+sidebar_current: "docs-internals-provider-guide-new-resource"
+description: |-
+  How to get started adding a new resource to an existing provider.
+---
+
+# Creating Resources
+
+Now that you’ve got a Provider, it’s time to add some Resources to that
+Provider to make it actually _do_ something. To do this, you need to write some
+code in the package your Provider is defined in.
+
+In the built-in plugins for Terraform that are shipped as part of the codebase,
+we have a file naming guideline: we use `resource_{resource_name}.go` as the
+template to name our files, and tend to stick to  single Resource per file. For
+example, AWS has `resource_aws_instance.go`, because it uses the AWS API to
+manage the instance resource. If you want to contribute your code back to the
+Terraform repo, it makes things easier if you follow this template.
+
+Inside this file, you’re going to define a function that takes no arguments and
+returns a `*schema.Resource`.  While you can technically name your function
+anything, we recommend (especially if you plan on contributing your code back
+to the Terraform repo!) that you name it the camelCase version of the file
+name. For example, `resource_compute_instance.go` would contain
+`resourceComputeInstance`. We’ll be defining a _lot_ of functions in a
+Provider, and this kind of naming scheme really helps keep things organised and
+easy to find.
+
+## Registering the Resource With the Provider
+
+Now that you’ve got your function, switch back to your Provider definition, and
```

Comment:
As I said above, I don't think type of language is appropriate for technical documentation or guides.